src/app/dataset-block/metadata-block/components/block-navigation/block-navigation.component.ts

```diff
@@ -1,9 +1,47 @@
-import { ChangeDetectionStrategy, Component } from "@angular/core";
+import { DatasetHistoryUpdate } from "./../../../../dataset-view/dataset.subscriptions.interface";
+import { SupportedEvents } from "./../event-details/supported.events";
+import {
+    ChangeDetectionStrategy,
+    Component,
+    EventEmitter,
+    Input,
+    Output,
+} from "@angular/core";
+import { DatasetInfo } from "src/app/interface/navigation.interface";
 
 @Component({
     selector: "app-block-navigation",
     templateUrl: "./block-navigation.component.html",
     styleUrls: ["./block-navigation.component.sass"],
     changeDetection: ChangeDetectionStrategy.OnPush,
 })
-export class BlockNavigationComponent {}
+export class BlockNavigationComponent {
+    @Input() public datasetHistory: DatasetHistoryUpdate;
+    @Input() public currentBlockHash: string;
+    @Input() public datasetInfo: DatasetInfo;
+    @Output() onPageChangeEmit = new EventEmitter<{
+        currentPage: number;
+        isClick: boolean;
+    }>();
+    public searchHash = "";
+    public eventType = this.sortOptions[0];
+    public currentPage = 1;
+
+    public get sortOptions() {
+        return ["None", ...Object.keys(SupportedEvents)];
```

Comment:
`None` to app values or somewhere, no to magic constants

Reply:
This code was deleted